activesupport/test/testing/callbacks_test.rb

```diff
@@ -10,6 +10,30 @@ def after_teardown
   end
 end
 
+class CallbackHookOrderTest < ActiveSupport::TestCase
+  setup { @callback_hooks = [:setup] }
+
+  teardown { @callback_hooks << :teardown }
+
+  around do |_, test|
+    @callback_hooks << :before_around
+
+    test.call
+
+    @callback_hooks << :after_around
```

Comment:
Would this get executed if the tests fails?

Reply:
@rafaelfranca yes, it will. I've tested this locally, but I'm unsure of how to include a failing test in the test suite. How might this test code gracefully force and catch a failure?